lib/ddtrace/workers.rb

```diff
@@ -116,6 +103,21 @@ def enqueue_service(service)
         return if service == {} # no use to send this, not worth it
         @service_buffer.push(service)
       end
+
+      private
+
+      def perform
+        loop do
+          @back_off = callback_traces ? @flush_interval : [@back_off * BACK_OFF_RATIO, BACK_OFF_MAX].min
+
+          callback_services
+
+          @mutex.synchronize do
+            return if !@run && @trace_buffer.empty? && @service_buffer.empty?
+            @shutdown.wait(@mutex, @back_off)
```

Comment:
Once the `@shutdown` conditional variable is signaled this thread awakes immediately so we can perform a last flush without having to wait for the back off to finish.

Reply:
Cool, this works thanks to `ConditionVariable` that releases the lock.